app/src/main/java/com/quran/labs/androidquran/ui/SurahAudioManager.java

```diff
@@ -0,0 +1,297 @@
+  private DefaultDownloadReceiver downloadReceiver;
+  private String basePath;
+  private List<QariItem> qariItems;
+  private SurahAdapter surahAdapter;
+  private int sheikhPosition = -1;
+
+
+  @Inject
+  QuranInfo quranInfo;
+  @Inject
+  QuranFileUtils quranFileUtils;
+  @Inject
+  AudioUtils audioUtils;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    QuranApplication quranApp = (QuranApplication) getApplication();
+    quranApp.getApplicationComponent().inject(this);
+    quranApp.refreshLocale(this, false);
+
+    super.onCreate(savedInstanceState);
+    final ActionBar ab = getSupportActionBar();
+    if (ab != null) {
+      ab.setTitle(R.string.audio_manager);
+      ab.setDisplayHomeAsUpEnabled(true);
+    }
+
+    setContentView(R.layout.activity_surah_audio_manager);
+
+    Intent intent = getIntent();
+    sheikhPosition = intent.getIntExtra("Sheikh", -1);
+
+    recyclerView = findViewById(R.id.recycler_view);
+    recyclerView.setHasFixedSize(true);
+    recyclerView.setLayoutManager(new LinearLayoutManager(this));
+    recyclerView.setItemAnimator(new DefaultItemAnimator());
+
+    progressBar = findViewById(R.id.progress);
+    qariItems = audioUtils.getQariList(this);
+    basePath = quranFileUtils.getQuranAudioDirectory(this);
+    surahAdapter = new SurahAdapter(qariItems);
+    recyclerView.setAdapter(surahAdapter);
+    getShuyookhData();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    downloadReceiver = new DefaultDownloadReceiver(this,
+        QuranDownloadService.DOWNLOAD_TYPE_AUDIO);
+    downloadReceiver.setCanCancelDownload(true);
+    LocalBroadcastManager.getInstance(this).registerReceiver(downloadReceiver,
+        new IntentFilter(QuranDownloadNotifier.ProgressIntent.INTENT_NAME));
+    downloadReceiver.setListener(this);
+  }
+
+  @Override
+  protected void onPause() {
+    downloadReceiver.setListener(null);
+    LocalBroadcastManager.getInstance(this).unregisterReceiver(downloadReceiver);
+    super.onPause();
+  }
+
+  @Override
+  protected void onDestroy() {
+    disposable.dispose();
+    super.onDestroy();
+  }
+
+  private DisposableSingleObserver<List<QariDownloadInfo>> mOnDownloadInfo =
+      new DisposableSingleObserver<List<QariDownloadInfo>>() {
+        @Override
+        public void onSuccess(List<QariDownloadInfo> downloadInfo) {
+          progressBar.setVisibility(View.GONE);
+          surahAdapter.setDownloadInfo(downloadInfo);
+          surahAdapter.notifyDataSetChanged();
+        }
+
+        @Override
+        public void onError(Throwable e) {
+        }
+      };
+
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    if (item.getItemId() == android.R.id.home) {
+      finish();
+      return true;
+    }
+    return super.onOptionsItemSelected(item);
+  }
+
+  private void getShuyookhData() {
+    if (disposable != null) {
+      disposable.dispose();
+      mOnDownloadInfo =
```

Comment:
can we rename this variable to not have an `m` at the start?

Reply:
actually why do we need this and the other one that's set on line 126? can we just inline one of them directly in the subscribe instead?